java/src/test/java/com/mlt/TestUtils.java

```diff
@@ -3,123 +3,111 @@
+        featureTables.length,
+        mvtLayers.size(),
+        "featureTables.length=" + featureTables.length + " mvtLayers.size()=" + mvtLayers.size());
     for (var i = 0; i < mvtLayers.size(); i++) {
-      var featureTable = featureTables[i];
       var mvtLayer = mvtLayers.get(i);
-      var mvtFeatures = mvtLayer.features();
+      var mvtFeatures = mvtLayers.get(i).features();
+      var featureTable = featureTables[i];
       var featureIterator = featureTable.iterator();
-
-      for (var j = 0; j < mvtFeatures.size(); j++) {
-        var mltFeature = featureIterator.next();
-        var mvtFeature =
-            isFeatureTableSorted
-                ? mvtFeatures.stream().filter(f -> f.id() == mltFeature.id()).findFirst().get()
-                : mvtFeatures.get(j);
-
-        assertEquals(mvtFeature.id(), mltFeature.id());
-
-        var mvtGeometry = mvtFeature.geometry();
-        var mltGeometry = mltFeature.geometry();
-        assertEquals(mvtGeometry, mltGeometry);
-
-        var mltProperties = mltFeature.properties();
-        for (var property : mltProperties.entrySet()) {
-          var mltPropertyKey = property.getKey();
-          var mltPropertyValue = property.getValue();
-          if (mltPropertyValue instanceof Map<?, ?>) {
-            /* Handle shared dictionary case -> currently only String is supported
-             * as nested property in the converter, so only handle this case */
-            var mvtProperties = mvtFeature.properties();
-            var nestedStringValues = (Map<String, String>) mltPropertyValue;
-            var mvtStringProperties =
-                mvtProperties.entrySet().stream()
-                    .filter(
-                        p -> p.getKey().contains(mltPropertyKey) && p.getValue() instanceof String)
-                    .toList();
-            // TODO: verify why mlt seems to have a property more than mvt on the
-            // name:* column in some tiles
-            for (var mvtProperty : mvtStringProperties) {
-              var mvtPropertyKey = mvtProperty.getKey();
-              var mvtPropertyValue = mvtProperty.getValue();
-              var mltValue = nestedStringValues.get(mvtPropertyKey);
-
-              if (mvtPropertyKey.equals("name:ja:rm")) {
-                // TODO: fix -> currently the converter can't handle a triple nested property name
-                System.out.println(
-                    "Skip verification for the name:ja:rm property name since it is currently"
-                        + " not supported in the converter.");
-                numErrors++;
-                continue;
-              }
-
-              assertEquals(mvtPropertyValue, mltValue);
-            }
-          } else {
-            assertEquals(mvtFeature.properties().get(mltPropertyKey), mltPropertyValue);
-          }
-        }
+      var mltFeatures = new ArrayList<Feature>();
+      while (featureIterator.hasNext()) {
+        mltFeatures.add(featureIterator.next());
```

Comment:
minor: could simplify to `var mltFeatures = new ArrayList<>(featureTable);` but also why not just do `numErrors += compareFeatures(featureTables[i], mvtFeatures, isFeatureTableSorted);` and skip making the copy?

Reply:
If I'm understanding right, it's not just a copy, it's a decoding into `List<Feature>` (the `featureIterator.next()` has to be called in order to trigger feature decoding). In general, there are two different decoding paths and they are both turned into `List<Feature>` here in order to have the same code run asserts on the output of the two paths.